python/pyspark/sql/connect/udf.py

```diff
@@ -77,6 +79,17 @@ def _create_py_udf(
     eval_type: int = PythonEvalType.SQL_BATCHED_UDF
 
     if is_arrow_enabled:
+        try:
+            require_minimum_pandas_version()
+            require_minimum_pyarrow_version()
+        except ImportError:
+            is_arrow_enabled = False
+            warnings.warn(
+                "Arrow optimization failed to enable because PyArrow is not installed. "
```

Comment:
nit: Looks like we also need Pandas here
```suggestion
                "Arrow optimization failed to enable because PyArrow or Pandas is not installed. "
```